p2p/status.go

```diff
@@ -0,0 +1,126 @@
+// Copyright 2019 ChainSafe Systems (ON) Corp.
+// This file is part of gossamer.
+//
+// The gossamer library is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Lesser General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// The gossamer library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Lesser General Public License for more detailg.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with the gossamer library. If not, see <http://www.gnu.org/licenses/>.
+
+package p2p
+
+import (
+	"time"
+
+	"github.com/ChainSafe/gossamer/common"
+	log "github.com/ChainSafe/log15"
+	"github.com/libp2p/go-libp2p-core/network"
+	"github.com/libp2p/go-libp2p-core/peer"
+)
+
+// SendStatusInterval is the time between sending status messages
+const SendStatusInterval = 5 * time.Minute
+
+// TODO: generate host status message
+var hostStatus = &StatusMessage{
+	ProtocolVersion:     0,
+	MinSupportedVersion: 0,
+	Roles:               0,
+	BestBlockNumber:     0,
+	BestBlockHash:       common.Hash{0x00},
+	GenesisHash:         common.Hash{0x00},
+	ChainStatus:         []byte{0},
+}
+
+// status submodule
+type status struct {
+	host        *host
+	confirmed   map[peer.ID]bool
+	lastMessage map[peer.ID]*StatusMessage
```

Comment:
Just a heads up, storing the status message here in preparation for storing its contents in state.

Reply:
Might be able to just expose a method to get a slice of `PeerInfo` for each confirmed connected peer.